flink-ai-flow/lib/notification_service/notification_service/event_storage.py

```diff
@@ -72,9 +74,75 @@ def list_all_events_from_id(self, id: int):
             if event.id > id:
                 res.append(event)
         return res
+    
+    def get_latest_version(self, key: str = None):
+        res = []
+        for event in self.store:
+            if event.key == key:
+                res.append(event.version)
+        res.sort()
+        return res[0] if res else 0
 
     def clean_up(self):
         self.store.clear()
         self.kv.clear()
         self.max_id = 0
 
+
+class MongoEventStorage(BaseEventStorage):
+    def __init__(self, *args, **kwargs):
+        self.db_conn = self.setup_connection(**kwargs)
+        self.server_id = str(uuid.uuid4())
+    
+    def setup_connection(self, **kwargs):
+        db_conf = {
+            "host": kwargs.get("host"),
+            "port": kwargs.get("port"),
+            "username": kwargs.get("username"),
+            "password": kwargs.get("password"),
+            "db": kwargs.get("db"),
+        }
+        # If ignore the authSource, the authentication will be failed.
+        conn_uri = "mongodb://{username}:{password}@{host}:{port}/{db}?authSource=admin".format(**db_conf)
+        db_conf["host"] = conn_uri
+        print("#####mongo db conf: {}".format(str(db_conf)))
```

Comment:
This print could be removed or recommend to use logging to log the MongoDB configuration.